### DISCUSSION TITLE
tutorials/intro-typescript/15-wrapping-up

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#124] [course feedback]
# tutorials/intro-typescript/15-wrapping-up

https://www.apollographql.com/tutorials/intro-typescript/15-wrapping-up

**Comment by jmacjamf** [comment]
when

**Reply by jmacjamf** [reply]
Getting error on the last update to resolvers.ts file for:
 AddItemsToPlaylistPayload: {
      playlist: ({ playlistId}, _, { dataSources }) => {
        return dataSources.spotifyAPI.getPlaylist(playlistId);
      },
    },

Binding element 'dataSources' implicitly has an 'any' type.
"Parameter '_' implicitly has an 'any' type.",
Binding element 'playlistId' implicitly has an 'any' type.
"message": "Object literal may only specify known properties, but 'AddItemsToPlaylistPayload' does not exist in type 'MutationResolvers<DataSourceContext, {}>'. Did you mean to write 'addItemsToPlaylist'?",

**Reply by jmacjamf** [reply]
Well, I fixed it.  I had the AddItemsToPlaylistPayload resolver NESTED inside the Mutation section, instead of separated.

**Reply by mabuyo** [reply]
Nice, glad you were able to catch that! 🙌🏽
